### PR TITLE
Concattenating empty strings for the purpose of getting a string representation has poor performance. Replacing where necessary.

### PR DESCRIPTION
**Pull request by KoenDG**
This very small PR is related to issue #24226 it is small exactly to keep it readable.

There is a single file with a large diff in this PR. I'm going to assume that Intellij did this in order to enforce the `.editorconfig` settings, which it picks up on automatically.

The file in question has a lot of trailing whitespace, which has now been removed, It seems this is not picked up by checkstyle. So if this isn't an issue, I'll revert those changes and focus only on the string concatenation.

As the title states, this PR is about code that does `field + ""` in order to get a string representation of said field. This was only necessary before Java 1.5 if I recall correctly.

Also, situations where primitive types are concatenated with an empty string have been replaced since this is not as performant as calling `String.valueOf()` on the field.

===========

1. Contributor agreement: Has been signed

2. Contributor guidelines: I have read them.

3. Gradle check: build successful.

4. PR is against master

5. The PR is not OS specific.

6. I'm not part of a class.

### CODE REVIEW

File: modules/lang-painless/src/main/java/org/elasticsearch/painless/DefMath.java [hunk 1]
```diff
@@ -354,33 +354,33 @@ private static Object rem(Object left, Object right) {
...
+
     private static boolean add(boolean a, boolean b) {
         throw new ClassCastException("Cannot apply [+] operation to type [boolean]");
     }
-    
+
     private static Object add(Object left, Object right) {
         if (left instanceof String || right instanceof String) {
-            return "" + left + right;
+            return String.valueOf(left) + right;
```
Comment (KoenDG) [inline modules/lang-painless/src/main/java/org/elasticsearch/painless/DefMath.java]: This is the actual change in the file with all the trailing whitespace.

Comment (jasontedor) [inline modules/lang-painless/src/main/java/org/elasticsearch/painless/DefMath.java]: Please revert all other changes in this file.

Comment (KoenDG) [inline modules/lang-painless/src/main/java/org/elasticsearch/painless/DefMath.java]: Done.

File: modules/lang-painless/src/main/java/org/elasticsearch/painless/DefMath.java [hunk 2]
```diff
@@ -380,7 +380,7 @@ private static boolean add(boolean a, boolean b) {
     
     private static Object add(Object left, Object right) {
         if (left instanceof String || right instanceof String) {
-            return "" + left + right;
+            return String.valueOf(left) + right;
```
Comment (jasontedor) [inline modules/lang-painless/src/main/java/org/elasticsearch/painless/DefMath.java]: I think I'd prefer to see this written as:

```
if (left instanceof String) {
    return (String) left + right;
} else if (right instanceof String) {
    return left + (String) right;
} else if (left instanceof Number) {
.
.
.
```

File: modules/lang-painless/src/main/java/org/elasticsearch/painless/node/EBinary.java [hunk 3]
```diff
@@ -291,7 +291,7 @@ private void analyzeAdd(Locals variables) {
             } else if (sort == Sort.DOUBLE) {
                 constant = (double)left.constant + (double)right.constant;
             } else if (sort == Sort.STRING) {
-                constant = "" + left.constant + right.constant;
+                constant = String.valueOf(left.constant) + right.constant;
```
Comment (jasontedor) [inline modules/lang-painless/src/main/java/org/elasticsearch/painless/node/EBinary.java]: We know these are not null from the outer if-block; so:

```
constant = left.constant.toString() + right.constant.toString();
```

**Review by jasontedor — CHANGES_REQUESTED** [review]
Looks good, I left two suggestions.

File: modules/lang-painless/src/main/java/org/elasticsearch/painless/node/EBinary.java (hunk 3 above)
Comment (KoenDG) [inline modules/lang-painless/src/main/java/org/elasticsearch/painless/node/EBinary.java]: Both changes have been implemented.

**Comment by jasontedor** [comment]
test this please

**Comment by KoenDG** [comment]
Is this okay to merge?

**Comment by jasontedor** [comment]
Can you merge master into this branch, push here, and we will fire off another CI run?

**Comment by KoenDG** [comment]
@jasontedor done

**Comment by jasontedor** [comment]
retest this please

**Review by jasontedor — APPROVED** [review]
LGTM.

**Pull request merged**